src/components/Payroll/PayrollConfiguration/PayrollConfigurationPresentation.tsx

```diff
@@ -111,157 +116,178 @@ export const PayrollConfigurationPresentation = ({
   }
 
   return (
-    <Flex flexDirection="column" gap={16}>
-      <Flex justifyContent="space-between" alignItems="center">
-        <FlexItem>
-          <Heading as="h1">{t('pageTitle')}</Heading>
-          <Text>
-            <Trans
-              i18nKey="description"
-              t={t}
-              components={{ dateWrapper: <Text weight="bold" as="span" /> }}
-              values={getPayrollConfigurationTitle({ payPeriod, locale })}
-            />
-          </Text>
-        </FlexItem>
-        <Button
-          title={t('calculatePayrollTitle')}
-          onClick={onCalculatePayroll}
-          isDisabled={isPending}
+    <div ref={containerRef} className={styles.container}>
+      <Flex flexDirection="column" gap={16}>
+        <Flex
+          flexDirection={isDesktop ? 'row' : 'column'}
+          justifyContent={isDesktop ? 'space-between' : 'normal'}
+          alignItems={isDesktop ? 'center' : 'stretch'}
+          gap={isDesktop ? 0 : 16}
         >
-          {t('calculatePayroll')}
-        </Button>
-      </Flex>
-
-      {(alerts || payrollDeadlineNotice) && (
-        <Grid gap={16} gridTemplateColumns="1fr">
-          {payrollDeadlineNotice && (
-            <Alert status="info" label={payrollDeadlineNotice.label}>
-              {payrollDeadlineNotice.content}
-            </Alert>
-          )}
-          {alerts}
-        </Grid>
-      )}
-
-      {isPending ? (
-        <LoadingIndicator>
-          <Flex flexDirection="column" alignItems="center" gap={4}>
-            <LoadingSpinner size="lg" />
-            <Heading as="h4">{t('loadingTitle')}</Heading>
-            <Text>{t('loadingDescription')}</Text>
-          </Flex>
-        </LoadingIndicator>
-      ) : (
-        <>
-          <div className={styles.payrollBlockerContainer}>
-            {payrollBlockers.length > 0 && (
-              <PayrollBlockerAlerts
-                blockers={payrollBlockers}
-                onMultipleViewClick={onViewBlockers}
+          <FlexItem>
+            <Heading as="h1">{t('pageTitle')}</Heading>
+            <Text>
+              <Trans
+                i18nKey="description"
+                t={t}
+                components={{ dateWrapper: <Text weight="bold" as="span" /> }}
+                values={getPayrollConfigurationTitle({ payPeriod, locale })}
               />
+            </Text>
+          </FlexItem>
+          <FlexItem flexGrow={isDesktop ? 0 : 0}>
```

Comment:
Nit: looks like this could just resolve to the same value?


```suggestion
          <FlexItem flexGrow={0}>
```